recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,687 @@
+            # Start by checking if python2 can be found
+            python_exe = tools.which("python2")
+            if not python_exe:
+                # Fall back on regular python
+                python_exe = tools.which("python")
+
+            if not python_exe:
+                msg = ("Python2 must be available in PATH "
+                        "in order to build Qt WebEngine")
+                raise ConanInvalidConfiguration(msg)
+            # In any case, check its actual version for compatibility
+            from six import StringIO  # Python 2 and 3 compatible
+            mybuf = StringIO()
+            cmd_v = "{} --version".format(python_exe)
+            self.run(cmd_v, output=mybuf)
+            verstr = mybuf.getvalue().strip().split('Python ')[1]
+            if verstr.endswith('+'):
+                verstr = verstr[:-1]
+            version = tools.Version(verstr)
+            # >= 2.7.5 & < 3
+            v_min = "2.7.5"
+            v_max = "3.0.0"
+            if (version >= v_min) and (version < v_max):
+                msg = ("Found valid Python 2 required for QtWebengine:"
+                        " version={}, path={}".format(mybuf.getvalue(), python_exe))
+                self.output.success(msg)
+            else:
+                msg = ("Found Python 2 in path, but with invalid version {}"
+                        " (QtWebEngine requires >= {} & < "
+                        "{})".format(verstr, v_min, v_max))
+                raise ConanInvalidConfiguration(msg)
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            self.options.with_icu = False
+        if self.settings.compiler == "apple-clang":
+            if tools.Version(self.settings.compiler.version) < "10.0":
+                raise ConanInvalidConfiguration("Old versions of apple sdk are not supported by Qt (QTBUG-76777)")
+
+
+    def configure(self):
+        if conan_version < Version("1.20.0"):
+            raise ConanInvalidConfiguration("This recipe needs at least conan 1.20.0, please upgrade.")
+        if self.settings.os != 'Linux':
+        #     self.options.with_libiconv = False
+            self.options.with_fontconfig = False
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5.3":
+            self.options.with_mysql = False
+        if self.settings.os == "Windows":
+            self.options.with_mysql = False
+            if not self.options.shared and self.options.with_icu:
+                raise ConanInvalidConfiguration("icu option is not supported on windows in static build. see QTBUG-77120.")
+
+        if self.options.widgets and not self.options.GUI:
+            raise ConanInvalidConfiguration("using option qt:widgets without option qt:GUI is not possible. "
+                                            "You can either disable qt:widgets or enable qt:GUI")
+        if not self.options.GUI:
```

Comment:
I don't like to override options and forcing a new value, but here I see no other option, as those options are true by default. I recommend adding a self.output.warn, with a message explain "The option GUI requires the follow options as disabled: 'vulkan', ...." Something like that.

Reply:
I'm not sure this if bloc is really pertinent. qt without GUI module is a really niche case, which is probably not tested at all upstream.  consumers using option qt:GUI=False will probably need an important number of other tweaks.

Reply:
You can also delete these options, and use `get_safe`.